stdlib/REPL/src/REPL.jl

```diff
@@ -1388,4 +1392,62 @@ function run_frontend(repl::StreamREPL, backend::REPLBackendRef)
     nothing
 end
 
+module IPython
+
+using ..REPL
+
+__current_ast_transforms() = isdefined(Base, :active_repl_backend) ? Base.active_repl_backend.ast_transforms : REPL.repl_ast_transforms
+
+function repl_eval_counter(hp)
+    length(hp.history)-hp.start_idx
+end
+
+function out_transform(x, repl=Base.active_repl)
+    return quote
+        julia_prompt = $repl.interface.modes[1]
+        n = $repl_eval_counter(julia_prompt.hist)
+        mod = $REPL.active_module()
+        if !isdefined(mod, :Out)
+            setglobal!(mod, :Out, Dict{Int, Any}())
+        end
+        local __temp_val = $x # workaround https://github.com/JuliaLang/julia/issues/464511
```

Comment:
:eyes:

Reply:
Living in the future heh

Should be https://github.com/JuliaLang/julia/issues/46451